import java.util.InputMismatchException;
import java.util.Scanner;

public class Gym {
	public static void main(String[]args) {
		
	try {
		Scanner scanner = new Scanner(System.in);
		
		System.out.println("Enter a number to divide:");
		int x = scanner.nextInt();
		
		System.out.println("Enter a number to divide by:");
		int y = scanner.nextInt();
		
		int z = x/y;
		
		System.out.println("result: "+ z);
	}
	catch(ArithmeticException e) {
		System.out.println("this proves there have been arithmetic exception");
	}
	catch(InputMismatchException e) {
		System.out.println("There have been mismatch input ");
	}
	catch(Exception i) {
		System.out.println("This can handle the rest of the exception");
	}
	finally {
		System.out.println("This will always print ");
	}
	}	
}

